src/clj/game/cards/icebreakers.clj

```diff
@@ -884,6 +905,48 @@
+                 :effect (req (let [selected-virus target
+                                    yusuf card
+                                    counters (get-in selected-virus [:counter :virus] 0)]
+                                (resolve-ability state side
+                                                 {:prompt "Spend how many counters?"
+                                                  :choices {:number (req counters)
+                                                            :default (req 1)}
+                                                  :effect (req (let [cost target]
+                                                                 (resolve-ability
+                                                                   state side
+                                                                   {:counter-cost [:virus cost]
+                                                                    :effect (effect (pump yusuf cost)
+                                                                                    (system-msg (str "spends " cost (pluralize " counter" cost) " from " (:title selected-virus)
+                                                                                                     " to add " cost " strength to Yusuf")))}
+                                                                   selected-virus nil)))}
+                                                 yusuf nil)))}
+                {:label "Break barrier subroutine(s)"
+                 :prompt "Choose a card with virus counters"
+                 :choices {:req #(pos? (get-in % [:counter :virus] 0))}
+                 :effect (req (let [selected-virus target
+                                    yusuf card
+                                    counters (get-in selected-virus [:counter :virus] 0)]
+                                (resolve-ability state side
+                                                 {:prompt "Spend how many counters?"
+                                                  :choices {:number (req counters)
+                                                            :default (req 1)}
+                                                  :effect (req (let [cost target]
+                                                                 (resolve-ability
+                                                                   state side
+                                                                   {:counter-cost [:virus cost]
+                                                                    :effect (effect (system-msg (str "spends " cost (pluralize " counter" cost) " from " (:title selected-virus)
+                                                                                                     " to break " cost (pluralize " barrier subroutine" cost) " with Yusuf")))}
+                                                                   selected-virus nil)))}
+                                                 yusuf nil)))}]}
+
```

Comment:
My DRY sensense tingle on this card. Could abstract out both functions since they only differ on `(effect ...)` and `:label`.

Reply:
I tried, but because the inner effects need to be bound to variables from the outer scope, I couldn't find a way to make it happen. Help me, DRY Master!

Reply:
Merge it like this and I'll find some time to dry it up.